build-scripts/build-custom-app.sh

```diff
@@ -0,0 +1,234 @@
+
+
+type TProps = {
+  handleSearchInput: (term: string, filter?: string, filterValue?: string) => void;
+  searchInput: string;
+};
+
+export const Hero = ({ handleSearchInput, searchInput }: TProps) => {
+  return (
+    <div className="pb-6 text-white pt-[28px] sm:pt-[48px] md:pt-[80px] lg:pt-[100px] xl:pt-[140px]">
+      <div className="container">
+        <div className="flex flex-col items-center justify-center mb-6">
+		  <p className="font-medium tracking-slight text-lg lg:text-3xl" data-cy="intro-message">
+          ${custom_app_name} custom app
+        </p>
+        </div>
+        <div className="max-w-screen-md mx-auto mt-6">
+          <LandingSearchForm handleSearchInput={handleSearchInput} placeholder="Search full text of any document" input={searchInput} />
+        </div>
+      </div>
+    </div>
+  );
+};
+
+
+EOF
+
+# Define the content for the routes.json
+
+read -r -d '' ROUTES_JSON_CONTENT <<EOM
+
+[
+  {
+    "title": "Methodology",
+    "path": "methodology",
+    "contentPath": "methodology"
+  }
+]
+
+EOM
+
+# Define the content for the index.ts file
+
+read -r -d '' INDEX_TS_CONTENT <<EOM
+
+export { default as Analytics } from "./Analytics";
+export { default as Hero } from "./Hero";
+
+EOM
+
+# Define the path to the themes directory
+themes_path="themes"
+
+# Define the path to the e2e directory
+e2e_path="e2e"
+
+# Check if the themes directory exists
+if [[ ! -d ${themes_path} ]]; then
+	echo "The themes directory does not exist. Creating it now..."
+	mkdir "${themes_path}"
+fi
+
+# Define the full path to the new directory
+new_dir="${themes_path}/${custom_app_name}"
+# Define the path to the subdirectories
+components_dir="${new_dir}/components"
+layouts_dir="${new_dir}/layouts"
+pages_dir="${new_dir}/pages"
+styles_dir="${new_dir}/styles"
+tests_dir="${new_dir}/tests"
+e2e_test_dir="${e2e_path}/cypress/e2e/${custom_app_name}"
+
+# Check if the directory already exists
+if [[ -d ${new_dir} ]]; then
+	echo "You are trying to duplicate a Custom App, ${new_dir} already exists."
+else
+	printf "The custom app does not exist, creating it now with the required sub-directories and files..."
+	mkdir -p "${new_dir}" "${components_dir}" "${layouts_dir}" "${pages_dir}" "${styles_dir}" "${tests_dir}" "${e2e_test_dir}"
+	printf " -> Created custom app directory for %s: %s\n\n" "${custom_app_name}" "${new_dir}"
+
+	# Create the required files
+	touch "${components_dir}/index.ts"
+	touch "${components_dir}/Analytics.tsx"
+
+	echo "${ANALYTICS_CONTENT}" >"${components_dir}/Analytics.tsx"
+	echo "${INDEX_TS_CONTENT}" >"${components_dir}/index.ts"
+	cp -r "${themes_path}/cclw/components/LandingSearchForm.tsx" "${components_dir}/LandingSearchForm.tsx"
+	printf "%s" "${HERO_CONTENT}" >"${components_dir}/Hero.tsx"
+
+	touch "${pages_dir}/homepage.tsx"
+	touch "${pages_dir}/methodology.tsx"
+	echo "${HOMEPAGE_CONTENT}" >"${pages_dir}/homepage.tsx"
+	echo "${METHODOLOGY_PAGE_CONTENT}" >"${pages_dir}/methodology.tsx"
```

Comment:
Do we still need the methodology.tsx file if we are creating MethodologyLink.tsx and returning null in there?